solver/src/solver/http_solver/settlement.rs

```diff
@@ -153,19 +153,16 @@ fn match_prepared_and_settled_amms(
 }
 
 fn match_settled_prices(
-    prepared_tokens: &HashMap<String, H160>,
     executed_limit_orders: &[ExecutedLimitOrder],
     executed_amms: &[ExecutedAmm],
     solver_prices: HashMap<String, Price>,
 ) -> Result<HashMap<H160, U256>> {
     // Remove the indirection over the token string index from the solver prices.
     let solver_prices: HashMap<H160, Price> = solver_prices
         .into_iter()
-        .map(|(index, price)| {
-            let token = prepared_tokens
-                .get(&index)
-                .ok_or_else(|| anyhow!("invalid token {}", index))?;
-            Ok((*token, price))
+        .map(|(address_string, price)| {
+            let token = H160::from_str(&address_string).expect("token string not parsable.");
```

Comment:
I don't think we should panic in case of failing to parse the result from the solver. This could make a bug in the HTTP solver crash the driver.

Reply:
This is also no longer relevant as we use H160 for this now.